cmake/src.cmake

```diff
@@ -53,10 +64,13 @@ else()
 endif()
 
 if(HDF5_FOUND)
-  target_link_libraries(matio
-    PUBLIC HDF5::HDF5)
-  target_link_libraries(matio-static
-    PUBLIC HDF5::HDF5)
+  if(WIN32)
+    target_link_libraries(matio PUBLIC hdf5::hdf5-shared)
+    target_link_libraries(matio-static PUBLIC hdf5::hdf5-static)
+  else()
+      target_link_libraries(matio PUBLIC HDF5::HDF5)
```

Comment:
yep, I just reused the target created in `thirdParties.cmake`, that's somehting that we can change later. However, since we have to set whether we use the static or the dynamic library before calling `find_package(HDF5)`, I'm affraid we cannot define both.